Sources/SPMTestSupport/MockPackageRegistryURLProtocol.swift

```diff
@@ -0,0 +1,177 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
+import TSCBasic
+
+public final class MockPackageRegistryURLProtocol: URLProtocol {
```

Comment:
does this needs to be in SPMTestSupport, or can it be in the module tests?

Reply:
It can be in the module tests, but I thought SPMTestSupport would was a better place for this. There's less of an argument for that if `HTTPClient` handlers are more likely to be used, though.

Reply:
I tend to think the only code that would use it would be the registry tests, so putting it there seems better to me. that said, could go either way, so up to you.

Reply:
Mooted with 1ba971a6